client/src/components/UserList.js

```diff
@@ -1,44 +1,42 @@
 import React from 'react';
 import axios from 'axios';
 import User from './User';
+import RequireAuth from './auth/RequireAuth';
 
-export default class UserList extends React.Component {
+class UserList extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       users: [],
     };
   }
   componentDidMount() {
-    const token = localStorage.getItem('user-token');
-    if (!token) return this.props.history.push('/signin');
-    const config = {
-      headers: {
-        Authorization: `Bearer ${token}`,
-      },
-    };
+    // Authorization header attached by RequireAuth HOC, which also handles checking for token and changing base URL.
     axios
-      .get('http://localhost:3030/api/users', config)
+      .get('/users')
       .then(res => this.setState({ users: res.data.users }))
       .catch(msg => {
-        this.props.displayError(msg.response ? msg.response.data.error : 'There was an error while retrieving users.');
+        if (msg.response){
+          if (msg.response.status === 401 || msg.response.status === 403) localStorage.removeItem('user-token');
+          this.props.displayError(msg.response.data.error);
+        } else this.props.displayError('There was an error while retrieving users.');
         this.props.history.push('/signin');
       });
   }
-  handleLogout = ev => {
-    localStorage.removeItem('user-token');
-    this.props.history.push('/signin');
-  };
   render() {
     return (
       <ul className="users-list">
+        <li class="labels">
```

Comment:
This `class` should be `className`, of course, since `class` is a reserved keyword in JavaScript.